task/install: run 'yum ... install' the right way

tchaikov opened the pull request:
after cd1cef6, we install rpm packages like

2018-10-25T06:31:33.578 INFO:teuthology.orchestra.run.smithi076:Running:
"sudo yum -y '' install '' '' libcephfs_jni1"
...
2018-10-25T06:31:34.180
INFO:teuthology.orchestra.run.smithi076.stderr:No such command: . Please
use /bin/yum --help
2018-10-25T06:31:34.193 DEBUG:teuthology.orchestra.run:got remote
process result: 1

which fails the qa run on rpm based distros.

in this change, the command line args are passed in itertools.chain(),
because orchestra.run.quote() takes an iterable which contains string or
Raw.

Signed-off-by: Kefu Chai <kchai@redhat.com>

tchaikov commented:
this should address most of the failures in http://pulpito.ceph.com/kchai-2018-10-25_05:48:50-rados-wip-kefu-testing-2018-10-25-1035-distro-basic-smithi/

tchaikov commented on teuthology/task/install/rpm.py:
```diff
@@ -179,28 +175,21 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
-                      pkg_mng_subcommand_opts, pkg_mng_install_opts,
-                      pkg, run.Raw(';'),
-                      'fi']
+                args=chain(
```
@kshtsk could you please check `orchestra.run.quote()` to see if it accept a string like this?

kshtsk replied:
@tchaikov I don't think quote is used for bare multiline string, it is directed to the ssh session directly and works fine.

kshtsk replied:
@tchaikov 
```
>>> from teuthology.orchestra.remote import Remote as R
>>> R('127.0.0.1').run(args="""x=$(ls | grep 'teuthology')
... if test -e $x ; then
...     echo "File $x exists"
... fi""")
2018-10-25 14:44:00,891.891 INFO:teuthology.orchestra.run.127:Running: 'true'
2018-10-25 14:44:00,942.942 INFO:teuthology.orchestra.run.127:Running: 'x=$(ls | grep \'teuthology\')\nif test -e $x ; then\n\techo "File $x exists"\nfi'
2018-10-25 14:44:00,995.995 INFO:teuthology.orchestra.run.127.stdout:File teuthology.txt exists
RemoteProcess(client=<paramiko.client.SSHClient object at 0x1050f7ed0>, args='x=$(ls | grep \'teuthology\')\nif test -e $x ; then\n\techo "File $x exists"\nfi', check_status=True, hostname='127')
```

tchaikov replied:
makes sense! probably we can have a fix to improve the readability of `remote.run()` command. but i dare not do so at this moment, as it would imply large code churn. and we will need very careful about this sort of change.

kshtsk commented on teuthology/task/install/rpm.py:
```diff
@@ -179,28 +175,21 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
+                    pkg_install_cmd.format(pkg=pkg).split(), [run.Raw(';')],
+                    ['fi'])
             )
         if pkg is None:
```
Reverse logic, and same as above.
```
if pkg:
    if dist_release in ['opensuse', 'sle']:
        remote.run(args="""
            if test ! -e {package} ; then
                sudo zypper -n --no-gpg-checks install --capability --no-recommends {package}
            fi
        """.format(package=pkg)
    else:
        remote.run(args="""
            if test ! -e {package} ; then
                sudo yum -y install {package}
            fi
        """.format(package=pkg)
else:
    if dist_release in ['opensuse', 'sle']:
        remote.run(args="""
            sudo zypper -n --no-gpg-checks install --capability --no-recommends {package}
        """.format(package=cpack)
    else:
        remote.run(args="sudo yum -y install %s" % cpack)
```

tchaikov replied:
@kshtsk it's out of scope of this change. as i think the focus of it is not really refactoring, it's more about fixing the rpm install issue on centos. we can always fix the reverse logic in a follow-up PR.

smithfarm replied:
I agree with @tchaikov - this fix LGTM

kshtsk replied:
I agree too

tchaikov commented:
@kshtsk sorry for merging it without addressing your comments. and thanks for your review. i just wanted a minimal fix so we can continue the qa tests before we have an optimal solution. so i jumped the gun and merged it.

kshtsk commented:
@tchaikov no problem, thanks for providing the fix fast

tchaikov commented:
gents, #1230 is a follow-up PR for the discussion here.
